src/AiClient.php

```diff
@@ -0,0 +1,373 @@
+        }
+
+        // Validate that the model supports the requested capability
+        $supportedCapabilities = $model->metadata()->getSupportedCapabilities();
+        $supportsCapability = false;
+        foreach ($supportedCapabilities as $supportedCapability) {
+            if ($supportedCapability->equals($capability)) {
+                $supportsCapability = true;
+                break;
+            }
+        }
+
+        if (!$supportsCapability) {
+            throw new \InvalidArgumentException(
+                sprintf(
+                    'Model "%s" does not support the "%s" capability',
+                    $model->metadata()->getId(),
+                    $capability->value
+                )
+            );
+        }
+
+        // Route to the appropriate method based on capability
+        if ($capability->isTextGeneration()) {
+            return self::generateTextResult($prompt, $model);
+        }
+
+        if ($capability->isImageGeneration()) {
+            return self::generateImageResult($prompt, $model);
+        }
+
+        if ($capability->isTextToSpeechConversion()) {
+            return self::convertTextToSpeechResult($prompt, $model);
+        }
+
+        if ($capability->isSpeechGeneration()) {
+            return self::generateSpeechResult($prompt, $model);
+        }
+
+        throw new \InvalidArgumentException(
+            sprintf('Capability "%s" is not yet supported for generation', $capability->value)
+        );
+    }
+
+    /**
+     * Creates a new message builder for fluent API usage.
+     *
+     * This method will be implemented once MessageBuilder is available.
+     * MessageBuilder will provide a fluent interface for constructing complex
+     * messages with multiple parts, attachments, and metadata.
+     *
+     * @since n.e.x.t
+     *
+     * @param string|null $text Optional initial message text.
+     * @return object MessageBuilder instance (type will be updated when MessageBuilder is available).
+     *
+     * @throws \RuntimeException When MessageBuilder is not yet available.
+     */
+    public static function message(?string $text = null)
+    {
+        throw new \RuntimeException(
+            'MessageBuilder is not yet available. This method depends on builder infrastructure. ' .
+            'Use direct generation methods (generateTextResult, generateImageResult, etc.) for now.'
+        );
+    }
+
+    /**
+     * Generates text using the traditional API approach.
+     *
+     * @since n.e.x.t
+     *
+     * @param Prompt $prompt The prompt content.
+     * @param ModelInterface|null $model Optional specific model to use.
+     * @return GenerativeAiResult The generation result.
+     *
+     * @throws \InvalidArgumentException If the prompt format is invalid.
+     * @throws \RuntimeException If no suitable model is found.
+     */
+    public static function generateTextResult($prompt, ?ModelInterface $model = null): GenerativeAiResult
```

Comment:
The exact feedback that I provided for `generateResult` applies to this method too.